app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
more like `listen 'SIGUSR1' { JobExecution.wait_for_jobs_to_finish; Process.kill $$, 'SIGUSR2' }` ... but that can be it's own PR if anyone wants that ...